packages/core-js/utils/comparator.js

```diff
@@ -0,0 +1,84 @@
+function compareStorageStructs({ previousStructsMap, currentStructsMap }) {
```

Comment:
This function finds the three kind of mutations that can happen in a struct. 
- Additions (valid mutation)
- Removals
- Modifications

Reply:
I think this file should go in the `ast.js` util or we should start having an `ast/` folder in the utils with specialized files.

Reply:
done